### DISCUSSION TITLE
Is there a defined spec for Workflow Syntax

### DISCUSSION THREAD

**Discussion by StephenHodgson** [community/community#139194] [Actions] [answered]
### Select Topic Area

Question

### Body
Is there a defined spec for [Workflow Syntax](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions)?
Ideally this would be in machine readable format (json/yaml)

But it'd be nice to programmatically generate libraries for compliance on the workflow syntax for generating workflows.

**Comment by ChristopherHX** [comment, marked as answer]
This one https://github.com/actions/languageservices/blob/main/workflow-parser/src/workflow-v1.0.json
There is one for action metadata as well somewhere under https://github.com/actions/runner


_If you are not looking for .net/javascript, I made a simple go syntax checker(it has defects, that for shure): https://github.com/nektos/act/tree/master/pkg/schema as well, since the go port is not public_
